drivers/include/periph/wdt.h

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (C) 2017 Technische Universität Berlin
+ *           (C) 2019 Inria
+ *           (C) 2019 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_periph_wdt WDT
+ * @ingroup     drivers_periph
+ * @brief       watchdog timer peripheral driver
+ *
+ * A watchdog timer (WDT) is an electronic or software timer that is
+ * used to detect and recover from unusual or suspect behaviour as well as
+ * software malfunctions. During normal operation the application will reset
+ * 'wdt_kick()' the timer counter preventing it from hitting 0. If due to
+ * software or hardware error the WDT counter isn't reset this will trigger
+ * corrective actions, i.e. a system reboot or callback (not supported by all
+ * platforms). 
+ *
+ * This interface defines two operation modes NORMAL and WINDOWED. In
+ * NORMAL operation if the WDT counter isn't reset before TIMEOUT then a
+ * reboot/callback is triggered. In WINDOWED operation if a reset isn't triggered
+ * or if it's triggered outside of the time window a reboot is triggered.
+ *
+ * Not all platforms support stopping the WDT, in some cases once enabled it
+ * can't be stopped so verify the specific platform implementation (e.g.
+ * only pause it in sleep modes).
+ *
+ * Since WDT upper and lower bounds are a per platform constraint they can be
+ * defined in periph_cpu.h and should be known at compile time. A distinction
+ * can be made between windowed WDT and normal WDT bounds, i.e.:
+ * 
+ * - NWDT_TIME_LOWER_BOUND & NWDT_TIME_UPPER_BOUND
+ * - WWDT_TIME_LOWER_BOUND & WWDT_TIME_UPPER_BOUND
+ *
+ * NORMAL operation
+ * ================
+ *
+ *
+ *    wdt_setup_reboot(0, MAX_TIME);
+ *    wdt_start();
+ * 
+ *    while (1) {
+ *        xtimer_usleep(MAX_TIME - x);
+ *        wdt_kick();
+ *    }
+ *
+* 0(ms)                                            MAX_TIME(ms)
+ * |----------------------------------------------------|
+ *
+ *                           x(ms)
+ * |---------------------------| - - - - - - -  - - - - |
+ *
+ *                             ^
+ *                             |
+ *                        wdt_kick()
+ * 
+ * WINDOWED operation
+ * ==================
+ *
+ *    wdt_setup_reboot(MIN_TIME, MAX_TIME);
+ *    wdt_start();
+ *
+ *    // x < (MAX_TIME - MIN_TIME)
```

Comment:
Please use C style comments also in documentation.

I think code snippet must be in specific code blocks:
```
'''c
/* C code (replace ticks by back ticks) */
'''
```

Reply:
If I put C style comments in a Doyxgen comment block it makes the whole code looks weird, I rather wither since it thinks the Doxygen comment block is ending there, so everything renders bad on IDE's

Reply:
That's why you need to put the snippet in a code block. There's a Markdown syntax for this.

Reply:
That doesn't work, RIOT code examples allways use "//" for comments within comments, you actually get compiler warning when compiling a file with "/* */" within another comment block

Reply:
Also I saw a lot of code uses this for enclosing code blocks:

```
 * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
...
 * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Is markdown better?

Reply:
I see and that's ok like this. But this is crap. Are the code comments really needed ?

Reply:
I guess I can move the comment out of the code snippet to have this:

```
 * In the code snippet and diagram ```time``` is an arbitrary value such that
 * ```time < MAX_TIME```.
 *
 * @verbatim
 * 0(ms)                                            MAX_TIME(ms)
 * |----------------------------------------------------|
 *
 *                          time(ms)
 * |---------------------------| - - - - - - -  - - - - |
 *
 *                             ^
 *                             |
 *                        wdt_kick()
 * @endverbatim
 * 
 * ~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
 * #include <stdio.h>
 *
 * #include "periph/wdt.h"
 * #include "xtimer.h"
 *
 * int main(void)
 * {
 *     wdt_setup_reboot(0, MAX_TIME);
 *     wdt_start();
 *
 *     while (1) {
 *         xtimer_usleep(time);
 *         wdt_kick();
 *     }
 *     return 0;
 * }
 * ~~~~~~~~~~~~~~~~~~~~~~~~
```
 Would that also be ok with you @benpicco ? (I changed x to time)

Reply:
Looks good to me.